Model/Export/Data/ImmediateProducts.php

```diff
@@ -0,0 +1,204 @@
+     * @param StockAttributeConfig $stockAttributeConfig
+     * @param AgeAttributeConfig $ageAttributeConfig
+     * @param ImageAttributeConfig $imageAttributeConfig
+     * @param CustomAttributeConfig $customAttributeConfig
+     * @param Json $json
+     * @param ResourceConnection $resource
+     * @param array $siteVariantPriceAttributes
+     * @param array $siteVariantStockAttributes
+     * @param array $siteVariantImageAttributes
+     * @param array $siteVariantAgeAttributes
+     */
+    public function __construct(
+        StoreDimensionProvider $dimensionProvider,
+        Full $indexerAction,
+        EngineProvider $engineProvider,
+        CollectionFactory $productCollectionFactory,
+        DataHandlerFactory $dataHandlerFactory,
+        GeneralConfig $generalConfig,
+        AttributeConfig $attributeConfig,
+        PricingAttributeConfig $pricingAttributeConfig,
+        StockAttributeConfig $stockAttributeConfig,
+        AgeAttributeConfig $ageAttributeConfig,
+        ImageAttributeConfig $imageAttributeConfig,
+        CustomAttributeConfig $customAttributeConfig,
+        Json $json,
+        ResourceConnection $resource,
+        array $siteVariantPriceAttributes = [],
+        array $siteVariantStockAttributes = [],
+        array $siteVariantImageAttributes = [],
+        array $siteVariantAgeAttributes = []
+    ) {
+        $this->dimensionProvider = $dimensionProvider;
+        $this->indexerAction = $indexerAction;
+        $this->engineProvider = $engineProvider;
+        $this->dataHandlerFactory = $dataHandlerFactory;
+        $this->productCollectionFactory = $productCollectionFactory;
+
+        parent::__construct(
+            $generalConfig,
+            $attributeConfig,
+            $pricingAttributeConfig,
+            $stockAttributeConfig,
+            $ageAttributeConfig,
+            $imageAttributeConfig,
+            $customAttributeConfig,
+            $json,
+            $resource,
+            $siteVariantPriceAttributes,
+            $siteVariantStockAttributes,
+            $siteVariantImageAttributes,
+            $siteVariantAgeAttributes
+        );
+        $this->resource = $resource;
+    }
+
+    /**
+     * Set Skus
+     *
+     * @param array $skus
+     */
+    public function setSkus(array $skus)
+    {
+        $this->skus = $skus;
+    }
+
+    /**
+     * Get Raw Product Data
+     *
+     * @param bool $isIncremental
+     * @param bool $isVariants
+     * @return array
+     */
+    protected function getRawProductData(bool $isIncremental, bool $isVariants) : array
+    {
+        $engine = $this->engineProvider->get();
+        if (!($engine instanceof \Aligent\FredhopperIndexer\Model\ResourceModel\Engine)) {
+            throw new \RuntimeException("Fredhopper is not configured as the search engine in Catalog Search");
+        }
+
+        /** @var \Magento\Framework\DB\Adapter\AdapterInterface $connection */
+        $connection = $this->resource->getConnection();
```

Comment:
Isn't there a core service class that gives product ids for a set of skus?

Reply:
`\Magento\Catalog\Model\ResourceModel\Product->getProductIdsBySkus(array $productSkuList)`

Reply:
So there is :+1: Updated